### PR TITLE
speeding up batch normalization

### PR DESCRIPTION
**Pull request by okuta**
This is speedup of batch normalization.

### CODE REVIEW

**Comment by unnonouno** [comment]
It looks good for me except a comment.

**Comment by okuta** [comment]
`shape[0]` is batch size. I expect loop unrolling by compiler, if batch size is small.
In most cases batch size is constant. Therefore, compiling the kernel only once in this code.

**Comment by unnonouno** [comment]
!! Cool idea!! So, please write a comment about loop unrolling.

**Comment by okuta** [comment]
Thank you for your advice.
I added comments.

**Comment by unnonouno** [comment]
:100: 
LGTM

**Comment by unnonouno** [comment]
Sorry, please hide `cuda.cutools` here:
https://github.com/pfnet/chainer/pull/7/files#diff-54bc089963209c5c3d69831ae45dd716R27

When CUDA is not enabled, it fails to import `batch_normalization`.
